mock/py/mockbuild/file_downloader.py

```diff
@@ -39,31 +40,30 @@ def get(cls, pkg_url_or_local_file):
         if not any([pkg.startswith(pfx) for pfx in url_prefixes]):
             log.debug("Local file: %s", pkg)
             return pkg
-
-        cls._initialize()
         try:
-            url = pkg
-            log.info('Fetching remote file %s', url)
-            req = requests.get(url)
-            req.raise_for_status()
-            if req.status_code != requests.codes.ok:
-                log.error("Can't download {}".format(url))
-                return False
+            log.info('Fetching remote file %s', pkg)
+            return cls._get_inner(pkg)
+        except requests.exceptions.RequestException as e:
+            log.error('Downloading error %s: %s', pkg, str(e))
+            return None
 
-            filename = urlsplit(req.url).path.rsplit('/', 1)[1]
-            if 'content-disposition' in req.headers:
-                _, params = cgi.parse_header(req.headers['content-disposition'])
-                if 'filename' in params and params['filename']:
-                    filename = params['filename']
-            pkg = cls.tmpdir + '/' + filename
-            with open(pkg, 'wb') as filed:
-                for chunk in req.iter_content(4096):
-                    filed.write(chunk)
-            cls.backmap[pkg] = url
-            return pkg
-        except Exception as err:  # pylint: disable=broad-except
-            log.error('Downloading error %s: %s', url, str(err))
-        return None
+    @classmethod
+    @backoff.on_exception(backoff.expo, requests.exceptions.RequestException, max_tries=3, max_time=10)
+    def _get_inner(cls, url):
+        req = requests.get(url)
+        req.raise_for_status()
+        cls._initialize()
```

Comment:
This can stay called outside, in get() (no need to call multiple times, it has no effect)